src/Model/Order/OrderTest.php

```diff
@@ -41,45 +44,75 @@ private function getAggregate($amount): Order
                         'countryCode' => "GB",
                     ],
                     'contactPerson' => "Jimmy Neutron",
-                    'hasPaid' => true,
+                    'isPaid' => true,
                     'amount' => $amount,
                 ]),
             ]
         );
         return $order;
     }
 
-    private function createOrder($amount)
+    public function test_order_amount_cannot_be_negative()
     {
+        $this->expectException(OrderAmountMustNotBeNegative::class);
+        $amount = -50;
         Order::create(
-            OrderId::generate(),
-            BusinessName::fromNative("Jimmy's car rental"),
-            BusinessAddress::fromNative([
+            NonNullOrderId::generate(),
+            NonNullBusinessName::fromNative("Jimmy's car rental"),
+            NonNullBusinessAddress::fromNative([
                 'addressLine1' => "No 3. Exeter Road",
                 'addressLine2' => "",
                 'town' => "Exeter",
                 'county' => "Devon",
                 'postcode' => "EX2 4QE",
                 'countryCode' => "GB",
             ]),
-            ContactPerson::fromNative("Jimmy Neutron"),
-            OrderIsPaid::true(),
-            OrderAmount::fromNative([
+            NonNullContactPerson::fromNative("Jimmy Neutron"),
+            NonNullOrderIsPaid::true(),
+            NonNullOrderAmount::fromNative([
                 "amount" => $amount,
                 "currency" => "gbp",
             ])
         );
     }
 
-    public function test_order_amount_cannot_be_negative()
+    public function test_order_amount_cannot_be_zero()
     {
-        $this->expectException(OrderAmountMustNotBeNegative::class);
-        $this->createOrder(-50);
+        $this->expectException(OrderAmountMustBeGreaterThanZero::class);
+        $amount = 0;
+        Order::create(
+            NonNullOrderId::generate(),
+            NonNullBusinessName::fromNative("Jimmy's car rental"),
+            NonNullBusinessAddress::fromNative([
+                'addressLine1' => "No 3. Exeter Road",
+                'addressLine2' => "",
+                'town' => "Exeter",
+                'county' => "Devon",
+                'postcode' => "EX2 4QE",
+                'countryCode' => "GB",
+            ]),
+            NonNullContactPerson::fromNative("Jimmy Neutron"),
+            NonNullOrderIsPaid::true(),
+            NonNullOrderAmount::fromNative([
+                "amount" => $amount,
+                "currency" => "gbp",
+            ])
+        );
     }
 
-    public function test_order_amount_cannot_be_zero()
+    public function test_order_should_record_the_time_it_was_paid()
```

Comment:
After we've updated the aggregate to get the datePaid property to be using the metadata from the OrderWasPaid event: we are going to need to use a `Clock` interface to test the system clock elements of that. You can ask me to come over to your desk to explain that bit when you get to it.

Reply:
We couldn't use the clock interface due to the inbuilt prooph init method calls createdAt dynamically within the function which is not testable.